src/Controller/UsersController.php

```diff
@@ -437,24 +437,69 @@ public function deleteUser(Request $request, PDO $db)
      * @throws Exception
      */
     public function setTrusted(Request $request, PDO $db)
+    {
+        $this->setAdminOnlyBooleanField($request, $db, 'trusted');
+    }
+
+    /**
+     * Allow users to be set as admin
+     *
+     * @param $request Request
+     * @param $db      PDO
+     *
+     * @throws Exception
+     */
+    public function setAdmin(Request $request, PDO $db)
+    {
+        $this->setAdminOnlyBooleanField($request, $db, 'admin');
+    }
+
+    /**
+     * Handles setting a boolean value that only an admin can set, since we have the same pattern happening 2x
+     * (and may add setting for the activity bool as well)
+     *
+     * @param $request Request
+     * @param $db      PDO
+     * @param $field   string
+     *
+     * @throws Exception
+     */
+    protected function setAdminOnlyBooleanField(Request $request, PDO $db, $field)
     {
         if (false == ($request->getUserId())) {
             throw new Exception("You must be logged in to change a user account", 401);
         }
 
+        if (!in_array($field, ['trusted', 'admin'])) {
+            throw new \InvalidArgumentException('Unknown boolean field change attempted', 501);
+        }
+
         $user_mapper = $this->getUserMapper($db, $request);
         if (!$user_mapper->isSiteAdmin($request->getUserId())) {
-            throw new Exception("You must be an admin to change a user's trusted state", 403);
+            throw new Exception("You must be an admin to change a user's $field state", 403);
```

Comment:
Probably better to use `sprintf` over string interpolation. It keeps it more readable.

Reply:
Fair enough. Just checked and we're using sprintf in plenty of other places here

Reply:
IMHO 

`"You must be an admin to change a user's $field state"`

Reads easier than 

`sprintf("You must be an admin to change a user's %s  state", $field);`

Reply:
Guess we need a style convention around this then. It's easier to search the codebase for sprintf than string interpolation, so I know the former exists a fair amount, but not sure about the latter. My personal pref is for interpolation because this is PHP rather than C and we're dealing with strings all the way around rather than e.g. formatting floats, but I'd rather codify one way or another than have to decide whether to revert the commit I just pushed :)